plugins/linux/ifplugin/interface_config.go

```diff
@@ -156,7 +156,14 @@ func (plugin *LinuxInterfaceConfigurator) ConfigureLinuxInterface(linuxIf *inter
 
 		return plugin.configureVethInterface(ifConfig, peerConfig)
 	case interfaces.LinuxInterfaces_AUTO_TAP:
-		return plugin.configureTapInterface(linuxIf.HostIfName, linuxIf)
+		var hostIfName string = ""
+		if linuxIf.Tap != nil  {
+			hostIfName = linuxIf.GetTap().GetTempIfName()
+		}
+		if hostIfName == "" {
```

Comment:
You can use `else` case instead of checking if the `hostIfName` is empty.

Reply:
But if tempIfName is empty the hostIfName will be empty even tap's hostIfName is defined.

Reply:
if linuxIf.Tap != nil && linuxIf.Tap.TempIfName != "" {
    return plugin.configureTapInterface(linuxIf.Tap.TempIfName, linuxIf)
} else {
    return plugin.configureTapInterface(linuxIf.HostIfName, linuxIf)
}

HostIfName is always set. In this case you can alse remove the first parameter (temp/host name can be read from linuxIf)